common/scala/src/main/scala/whisk/core/entity/Identity.scala

```diff
@@ -27,8 +27,15 @@ import whisk.core.database.MultipleReadersSingleWriterCache
 import whisk.core.database.NoDocumentException
 import whisk.core.entitlement.Privilege
 import whisk.core.entitlement.Privilege.Privilege
+import scala.util.Try
 
-protected[core] case class Identity(subject: Subject, namespace: EntityName, authkey: AuthKey, rights: Set[Privilege]) {
+case class UserLimits(invocationsPerMinute: Option[Int] = None, concurrentInvocations: Option[Int] = None, firesPerMinute: Option[Int] = None)
```

Comment:
should the user limits be itself the option rather than a class of fields that are all options?
i suppose this is more fine grained and gives more control - i wonder though if in general when changing throttles per user if we're not changing all these values at once.